measure_vram.py

```diff
@@ -0,0 +1,205 @@
+import torch
+import argparse
+import torch.optim as optim
+from datasets import load_dataset
+from torch.utils.data import DataLoader
+
+from data.collators import VQACollator
+from data.datasets import VQADataset
+from data.processors import get_image_processor, get_tokenizer
+from models.vision_language_model import VisionLanguageModel
+import models.config as config
+
+import os
+os.environ["TOKENIZERS_PARALLELISM"] = "false"
+
+torch.manual_seed(0)
+if torch.cuda.is_available():
+    torch.cuda.manual_seed_all(0)
+
+def measure_vram(args, vlm_cfg, train_cfg_defaults):
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    if not torch.cuda.is_available():
+        print("CUDA not available. VRAM measurement requires a CUDA-enabled GPU.")
+        return
+
+    # --- Model Initialization ---
+    if args.vlm_checkpoint_path:
+        print(f"Loading model from specified checkpoint: {args.vlm_checkpoint_path}")
+        model = VisionLanguageModel.from_pretrained(args.vlm_checkpoint_path)
+    else:
+        print(f"Initializing a new model (no specific checkpoint path provided via CLI).")
+        print(f"Using VLMConfig defaults: load_backbone_weights={vlm_cfg.vlm_load_backbone_weights}")
+        model = VisionLanguageModel(vlm_cfg, load_backbone=vlm_cfg.vlm_load_backbone_weights)
+    
+    model.to(device)
```

Comment:
Do you think we should also print out how much VRAM is needed for the model? This would be a very small addition to code, but later on if users wanted to play around with the architecture this would be helpful to know.